x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningUsageTransportAction.java

```diff
@@ -368,11 +370,16 @@ private static void updateStats(Map<String, Long> statsMap, Long value) {
     }
 
     private void addDeploymentStats(GetDeploymentStatsAction.Response response, Map<String, Object> inferenceUsage) {
+        TrainedModelAllocationMetadata trainedModelAllocationMetadata = TrainedModelAllocationMetadata.fromState(clusterService.state());
         StatsAccumulator modelSizes = new StatsAccumulator();
         double avgTimeSum = 0.0;
         StatsAccumulator nodeDistribution = new StatsAccumulator();
         for (var stats : response.getStats().results()) {
-            modelSizes.add(stats.getModelSize().getBytes());
+            TrainedModelAllocation allocation = trainedModelAllocationMetadata.getModelAllocation(stats.getModelId());
```

Comment:
The function `addTrainedModelStats` already adds a usage field `model_size_bytes` as map populate from the config:
 
`estimatedMemoryUsageBytes.add(trainedModelConfig.getModelSize());` 

I think `model_size_bytes` was added here just because it was available in the response. I think it would be neater to not set the field in `addTrainedModelStats` and call `GetTrainedModelStats` here instead of `GetDeploymentStatsAction` and populate the `model_size_bytes` from the stats as that is where we have the best information

Reply:
right, yes, that makes sense. I'll make the change.